test/test_main.py

```diff
@@ -81,6 +81,7 @@ def __init__(self, event_type, event_name, env_var=None,
         self.set_in_env = set_in_env
 
 
+@patch('ops.main.setup_root_logging', new=lambda *a, **kw: None)
```

Comment:
Would it actually be clearer to change when we would call setup_root_logging so that we don't trigger it always?
We can land this patch, but it feels like we need to then add the same check for anything that we add in a new test, so ideally it would be the default, and then we only trigger setup_root_logging explicitly.

Reply:
so the way i debugged this was by having `log._allowed = True`, an `if not _allowed: raise RuntimeError` in `log.setup_root_logging`, `test/__init__.py` setting `ops.log._allowed = False`, and then just `test_log`'s `setUp` and `tearDown` setting and clearing the flag.
It seemed to me to be alright for debugging but not shippable, but if I'm reading you right, you're saying it would be better to have something like that, perhaps instead of raising an exception, stopping it from being called?